content/posts/2022-04-11-postgresql-installation.md

```diff
@@ -0,0 +1,85 @@
+---
+title: 'تركيب قاعدة بيانات postgresql'
+date: 2022-04-11
+slug: 'postgreql-installation'
+template: 'post'
+categories:
+  - database administration
+tags:
+  - postgrseql
+  - database administration
+  - إدارة قواعد بيانات 
+thumbnail: '../thumbnails/postgresql.png'
+---
+
+## مقدمة بسيطة
+بسم الله الرحمن الرحيم
+
+تعد قاعدة بيانات `postgresql` من أشهر قواعد البيانات مفتوحة المصدر. 
+وهناك عدة طرق لتركيب هذا النوع من قواعد البيانات حيث تختلف الطريقة بناء على نظام التشغيل المستخدم. 
+في هذا الموضوع سوف نقوم بشرح طريقة التركيب على نظام `Linux` وبالاخص توزيعة `debian`.
+
+كما هو معلوم في عالم المصدر المفتوح ٬ من الممكن تحميل المصدر ثم القيام بناء الملفات التشغيلة من المصدر لكن الطريقة الأسهل هو تركيب قاعدة البيانات عن طريق الحزمة `package`. وهذا ماسوف نقوم بعمله في هذا الموضوع
+### ملاحظة مهمة
+من الأخطاء الشائعة هو عمل تركيب لقواعد البيانات من غير دراسة متطلبات التطبيقات التي سوف تعمل عليها حيث كل تطبيق قد يكون له احتياج خاص. فيجب دراسة متطلبات التطبيق ثم القيام بعمل التركيب بناء على المتطلبات.
+
+عمل تركيب لقاعدة البيانات بالاعدادت الافتراضية في الغالب  يكفي في التطبيقات الصغيرة لكن يجب تغير الاعدادات على حسب الاحتياج حيث اداء قاعدة البيانات سوف يتأثر بناء علي القيم المعطاة في الاعدادت.
+
+##  خطوات التركيب
+قبل البدء بالتركيب من المهم توضحية هو معرفة الاصدار الذي سوف يتم استخدامة حيث بعض الاصدارات مدعومة بشكل افتراضي في نظام التشغيل.
+
+لمعرفة الحزم المتوفرة في نظام التشغيل ، علينا زيارة موقع الرسمي لتوزيعة debain.
+https://www.debian.org/distrib/packages
+ثم البحث عن postgresql. 
+كما هو موضح في الصورة
+![package-search](../images/postgresql-installation/package-search.png "package-search")
+
+نلاحظ أن إصدار `postgresql` المتوفر في نظام تشغيل `debain bullseye` 
+هو الاصدار رقم `13`
+
+كل ماعلينا عملة لتركيب هذه الحزمة هو الامر التالي:
+```
+sudo apt install postgresql
```

Comment:
Debian do not come with sudo by default, maybe add 2 options to install using sudo or superuser or remove sudo